Tensile/KernelWriterAssembly.py

```diff
@@ -3108,22 +3108,30 @@ def allocateResources(self, kernel):
       kStr += ".if 0\n"
 
     # add offset to buffer
+    tmpOffset = self.sgprPool.checkOutAligned(2, 2, preventOverflow=0)
```

Comment:
I am not sure the difference but I do not see checkOut or checkOutAligned is used for sgpr.
Normally, self.getTmpSgpr is used instead.
Any specific reason you used checkOutAligned here instead of self.getTmpSgpr?

Reply:
checkOutAligned is in AsmRegisterPool.py:124. Since the two registers are used together for a 64-bit op, I think they need to be 64-bit aligned on gfx90a.

Reply:
It might not work if we call self.getTmpSgpr  twice in the same function.
If I remember correctly, calling getTmpSgpr did not work properly for some reason....
In graWorkGroup, we already have stmp = self.getTmpSgpr(4, 4).idx().
We might need to add 2 to this function instead of calling getTmpSgpr again.

Reply:
I am OK to use checkOutAligned if it is necessary.
I jut would like to know when we should use checkOut or Aligned for sgpr.

Reply:
Chatted with Koji about this one offline. Led to us discovering some additional updates that should be done on sgpr register allocation. I've posted an update for the sgpr allocations related to this change and will work on the others later in a separate PR. Thanks!